examples/client/test_object.c

```diff
@@ -214,35 +222,42 @@ static uint8_t prv_write(uint16_t instanceId,
 {
     prv_instance_t * targetP;
     int i;
-
+    char * tmp;
+    
     targetP = (prv_instance_t *)lwm2m_list_find(objectP->instanceList, instanceId);
     if (NULL == targetP) return COAP_404_NOT_FOUND;
 
     for (i = 0 ; i < numData ; i++)
     {
         switch (dataArray[i].id)
         {
-        case 1:
-        {
-            int64_t value;
-
-            if (1 != lwm2m_data_decode_int(dataArray + i, &value) || value < 0 || value > 0xFF)
-            {
-                return COAP_400_BAD_REQUEST;
-            }
-            targetP->test = (uint8_t)value;
-        }
-        break;
-        case 2:
-            return COAP_405_METHOD_NOT_ALLOWED;
-        case 3:
-            if (1 != lwm2m_data_decode_float(dataArray + i, &(targetP->dec)))
+            case 1:
             {
-                return COAP_400_BAD_REQUEST;
+                int64_t value;
+
+                if (1 != lwm2m_data_decode_int(dataArray + i, &value) || value < 0 || value > 0xFF)
+                {
+                    return COAP_400_BAD_REQUEST;
+                }
+                targetP->test = (uint8_t)value;
             }
             break;
-        default:
-            return COAP_404_NOT_FOUND;
+            case 2:
+                return COAP_405_METHOD_NOT_ALLOWED;
+            case 3:
+                if (1 != lwm2m_data_decode_float(dataArray + i, &(targetP->dec)))
+                {
+                    return COAP_400_BAD_REQUEST;
+                }
+                break;
+            case 4:
+                tmp = targetP->str;
+                targetP->str = lwm2m_malloc(dataArray[i].value.asBuffer.length + 1);
```

Comment:
Should make sure it is a string or opaque type first or dataArray[i].value.asBuffer.length may not be valid.

Reply:
Fixed